### ISSUE TITLE
OptimalBinning and user_splits

### ISSUE THREAD

**Issue by FedericoMontana** [guillermo-navas-palencia/optbinning#146]
Hi - I'm using OptimalBinning to create the splits using a dataset, and with those splits get the bins table from a different dataset. To do that, I'm using user_splits for the second dataset binning, expecting to get the same splits. I have noticed that in some cases the binning joins the splits indicated in user_splits, for example, the split is: -inf, 2, 3, 5, +inf, and when using that with user_splits it might merge the last 2: -inf, 2, 3, +inf. 
I even tried to use user_splits_fixed but it made things worse.

To summarize, by reading the documentation I'd expect that when using user_splits the binning is fully respected by the class, even when one of the bins has zero items.

Thank you!

**Comment by guillermo-navas-palencia** [comment]
Hi @FedericoMontana,

To apply user_splits_fixed effectively, you would need to deactivate other default constraints. These two changes are key: 
- monotonic_trend=None
- max_pvalue=None 

The algorithm cannot accept a bin with zero items because the WoE values are infinity (and other divergence metrics).
